allennlp/modules/elmo.py

```diff
@@ -681,3 +685,148 @@ def create_cached_cnn_embeddings(self, tokens: List[str]) -> None:
                                          weight=embedding.data,
                                          trainable=self._requires_grad,
                                          padding_index=0)
+
+
+class _ElmoSoftmax(torch.nn.Module):
+    """
+    Given a word :class:`Vocabulary` and the weights for the softmax
+    layer, return the log probabilities from the ELMo model. The `forward()`
+    method requires you to run :class:`_ElmoBiLm` and `batch_to_ids`
+    beforehand to calculate the probabilities.
+
+    Parameters
+    ----------
+    softmax_weight_file : ``str``, required
+        The .hdf5 file for the softmax weights.
+    softmax_vocab_file : ``str``, optional, (default None)
+        The .txt file for the vocabulary used for training ELMo.
+    chunk_size : ``int``, optional, (default 256)
+        The chunk size of the softmax layer. The default (256) takes about
+        10GB of memory!
+    """
+
+    def __init__(self,
+                 softmax_weight_file: str,
+                 softmax_vocab_file: str = None,
+                 chunk_size: int = 256):
+
+        super(_ElmoSoftmax, self).__init__()
+
+        self.softmax_weight_file = softmax_weight_file
+        self.softmax_vocab_file = softmax_vocab_file
+        self.chunk_size = chunk_size
+
+        if self.softmax_vocab_file:
```

Comment:
What is the purpose of loading the vocabulary in this module? It isn't used anywhere.